Add version flag to operator binary

vthapar opened the pull request:
Refer https://github.com/submariner-io/enhancements/pull/204

Fixes #2751

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our developer guide: https://submariner.io/development/
2. Ensure you have added the appropriate tests for your PR: https://submariner.io/development/code-review/#test-new-functionality
3. Read the code review guide to ease the review process: https://submariner.io/development/code-review/
4. If the PR is unfinished, mark it as a draft: https://submariner.io/development/code-review/#mark-work-in-progress-prs-as-drafts
5. If you are using CI to debug, use your private fork: https://submariner.io/development/code-review/#use-private-forks-for-debugging-prs-by-running-ci
6. Add labels to the PR as appropriate.

This template is based on the K8s/K8s template:

https://github.com/kubernetes/kubernetes/blob/master/.github/PULL_REQUEST_TEMPLATE.md
-->

vthapar committed:
Add version flag to operator binary

Refer https://github.com/submariner-io/enhancements/pull/204

Fixes #2751

Signed-off-by: Vishal Thapar <5137689+vthapar@users.noreply.github.com>

sridhargaddam commented on main.go:
```diff
@@ -51,27 +51,30 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/manager/signals"
 )
 
+const componentName = "submariner-operator"
```
Rather than redefining the component name, we can use the constant from https://github.com/submariner-io/submariner-operator/blob/ffeb0c063f9aafc0de6fe978d9b5564ed342fe89/pkg/names/names.go#L31

vthapar committed:
Use component names from admiral

Use admiral/pkg/names/names.go for component names for consistency
across all repos/components.

Signed-off-by: Vishal Thapar <5137689+vthapar@users.noreply.github.com>

tpantelis commented:
The `check_operator_version` script needs to be updated.
